Fix up of: Hide the mouse when screen curtain is on (PR #10182)

JulienCochuyt opened the pull request:
Only hide the mouse when screen curtain effectively turned on.
This prevents the mouse to disappear if the screen curtain could no be turned
on because the Windows Magnifier is started.

<!--
Please fill in the following template, for an explanation of the sections see:
https://github.com/nvaccess/nvda/wiki/Github-pull-request-template-explanation-and-examples
-->

### Link to issue number:

Fix up of PR #9982 

### Summary of the issue:

If the Windows Magnifier is on, Screen Curtain fails to turn on but the mouse is hidden.
To show the mouse again, one has to turn Windows Magnifier off then turn Screen Curtain on and off then turn Windows Magnifier back on.

### Description of how this pull request fixes the issue:

Hide the mouse only when Screen Curtain effectively turned on.

### Testing performed:

Checked the mouse is not hidden if Screen Curtain failed to turn on.
Checked the mouse is hidden if Screen Curtain succeeded to turn on.
Checked the mouse is shown again when Screen Curtain is turned off.

### Known issues with pull request:

### Change log entry:

N/A

JulienCochuyt committed:
Fix up of: Hide the mouse when screen curtain is on (PR #10182)

Only hide the mouse when screen curtain effectively turned on.
This prevents the mouse to disappear if the screen curtain could no be turned
on because the Windows Magnifier is started.

leonardder commented on source/visionEnhancementProviders/screenCurtain.py:
```diff
@@ -80,8 +80,8 @@ def canStart(cls):
+		Magnification.MagShowSystemCursor(False)
 
 	def terminate(self):
 		super(VisionEnhancementProvider, self).terminate()
```
Could you please store this line in a try block, and the call that restores the mouse in a finally block? This ensures that it is always executed.

JulienCochuyt replied:
> Could you please store this line in a try block, and the call that restores the mouse in a finally block? This ensures that it is always executed.

Belt and suspenders.
Done.

LeonarddeR reviewed (changes requested):
Good catch, thanks @JulienCochuyt

JulienCochuyt committed:
Fix up of: Hide the mouse when screen curtain is on (PR #10182)

Review action https://github.com/nvaccess/nvda/pull/10211#discussion_r324069933

leonardder commented on source/visionEnhancementProviders/screenCurtain.py:
```diff
@@ -80,13 +80,17 @@ def canStart(cls):
+		try:
+			super(VisionEnhancementProvider, self).terminate()
+			Magnification.MagUninitialize()
+		except:  # noqa: E722
```
These two lines don't have to be there. not specifying an except in a try/finally will just raise.

Having said that, the only exception I expect is Windows Error.

leonardder commented on source/visionEnhancementProviders/screenCurtain.py:
```diff
@@ -80,13 +80,17 @@ def canStart(cls):
+		except:  # noqa: E722
+			raise
+		finally:
+			Magnification.MagShowSystemCursor(True)
```
I always try to make the order of __init__ the reverse of terminate. Now, in both __init_- and terminate, we either initialize/unlitialize the engine and then call ShowSystemCursor. IOI would have expected that the latter failed, as you already called MagUNitialize before calling MagShowSYstemCursor.

I intended something like this, if you agree:

```
	def terminate(self):
		try:
			super().terminate()  # Feel free to make this a new style super call
		finally:
			Magnification.MagShowSystemCursor(True)
			Magnification.MagUninitialize()
```

JulienCochuyt replied:
Of course, my bad, it's much better that way.

JulienCochuyt committed:
Fix up of: Hide the mouse when screen curtain is on (PR #10182)

Review action https://github.com/nvaccess/nvda/pull/10211#discussion_r324119573
